control-plane/agents/src/bin/core/node/specs.rs

```diff
@@ -183,4 +185,94 @@ impl ResourceSpecsLocked {
         registry.store_obj(&drained_node_spec).await?;
         Ok(drained_node_spec)
     }
+
+    /// Add the draining volume to the node spec for checking shutdown nexuses.
+    pub(crate) async fn add_node_draining_volume(
+        &self,
+        registry: &Registry,
+        node_id: &NodeId,
+        volume_id: &VolumeId,
+    ) -> Result<NodeSpec, SvcError> {
+        let node = self.node_rsc(node_id)?;
+        let drained_node_spec = {
+            let mut locked_node = node.lock();
+            locked_node.add_draining_volume(volume_id);
+            locked_node.clone()
+        };
+        registry.store_obj(&drained_node_spec).await?;
+        Ok(drained_node_spec)
+    }
+
+    /// Get the vector of draining volumes on this node.
+    pub(crate) async fn get_node_draining_volumes(
```

Comment:
```suggestion
    pub(crate) async fn node_draining_volumes(
```
getters in rust don't tend to have "get_" prefix

Reply:
fixed